javascript/javascript1/week2/index.js

```diff
@@ -0,0 +1,234 @@
+const Male = "male";
+const Female = "female";
+
+function getFullname(firstName, sirName, useFormalName = false, gender = Male) {
+  if (arguments.length < 2) {
+    console.warn("First name and last name must be given");
+    return;
+  }
+  if (typeof firstName != "string" || typeof sirName != "string") {
+    console.warn("First name and last name must be of type string");
+    return;
+  }
+  //remove empty spaces within the name
+  if (firstName.length > 0) {
+    //firstName = firstName.Trim();
+    firstName = firstName.replace(/\s/g, "");
+  }
+  if (sirName.length > 0) {
+    //sirName = sirName.trim();
+    sirName = sirName.replace(/\s/g, "");
+  }
+  if (firstName === "" && sirName === "") {
+    //throw new console.error("First name and sir name should not be empty");
+    console.warn("First name and sir name should not be empty");
+    //throw "First name and sir name should not be empty";
+    return;
+  }
+  if (!(useFormalName === true || useFormalName === false)) {
+    //throw "Invalid parameter, bolean expected";
+    console.warn("Invalid parameter, bolean expected");
+    return;
+  }
+
+  if (useFormalName) {
+    if (gender === Male) {
+      return `Lord ${firstName} ${sirName}`;
+    } else if (gender === Female) {
+      return `Lady ${firstName} ${sirName}`;
+    }
+  }
+  return `${firstName} ${sirName}`;
+}
+
+const fullname1 = getFullname("Reenu", "Abraham");
+if (fullname1 != undefined) {
```

Comment:
In an actual scenario , where I have to use the output string somewhere else, how it will be if I return error message? How will I differentiate error message from successful full name returned. That's why I used console.warn ... throw error will stop the execution of the rest of the code. how is exception handling done in JS?

Reply:
But in this particular case, you don't want to return anything except an error message and yes you want to stop the execution of the function if the first name and/or surname is not there or it is not a string.